src/averageAges.js

```diff
@@ -37,7 +40,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const females = withChildren
+    ? people.filter(female =>
+      people.some(child => child.mother === female.name) && female.sex === 'f'
```

Comment:
```suggestion
      people.some(child => child.mother === female.name)
```

Reply:
I think the second check is unnecessary, because if `child.mother === female.name`, variable female definitely is a woman and is the mother